relations/src/r1cs/constraint_system.rs

```diff
@@ -767,6 +767,14 @@ impl<F: Field> ConstraintSystemRef<F> {
         }
     }
 
+    /// Consumes self to return the inner `ConstraintSystem<F>`. Returns
+    /// `None` if `Self::CS` is `None` or if any other references to
+    /// `Self::CS` exist.  
+    pub fn into_inner(self) -> Option<ConstraintSystem<F>> {
+        self.inner()
+            .map(|cs| Rc::try_unwrap(cs.clone()).unwrap().into_inner())
```

Comment:
Sorry I meant something like this, the goal being to avoid panics if more than one reference to the inner `cs` exists.

```suggestion
		match self {
			Self::CS(a) => Rc::try_unwrap(a).ok().map(|s| s.into_inner()),
			Self::None => None,
		}
```